codecs/wp2/enc/wp2_enc.cpp

```diff
@@ -7,7 +7,31 @@ using namespace emscripten;
 
 thread_local const val Uint8Array = val::global("Uint8Array");
 
-val encode(std::string image_in, int image_width, int image_height, WP2::EncoderConfig config) {
+struct WP2Options {
+  float quality;
+  float alpha_quality;
+  int speed;
+  int pass;
+  int uv_mode;
+  float sns;
+  int csp_type;
+  int error_diffusion;
+  bool use_random_matrix;
+};
```

Comment:
I created my own struct for this, otherwise it becomes a pain in the arse when it comes to enums.